File: RadVolontera/UI/radvolontera_admin/lib/models/payment/payment.dart
import 'dart:ffi';


import 'package:json_annotation/json_annotation.dart';

import '../account/account.dart';

part 'payment.g.dart';

@JsonSerializable()
class PaymentModel {
  int? id;
  String? notes;
  double amount;
  int? month;
  String ? monthName;
  int? year;
  String? studentId;
  AccountModel? student;
 
 PaymentModel(this.id, this.notes, this.amount, this.month,this.year, this.studentId, this.student,this.monthName);
      /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory PaymentModel.fromJson(Map<String, dynamic> json) => _$PaymentModelFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$PaymentModelToJson(this);
}

File: RadVolontera/RadVolontera.Models/Payment/Payment.cs
using RadVolontera.Models.Account;
using RadVolontera.Models.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RadVolontera.Models.Payment
{
    public  class Payment
    {
        public long Id { get; set; }
        public string Notes { get; set; }
        public double Amount { get; set; }
        public Month Month { get; set; }
        public string MonthName { get; set; }
        public int Year { get; set; }
        public string StudentId { get; set; }
        public UserResponse Student { get; set; }
    }
}
